src/main/java/org/italiangrid/storm/webdav/server/servlet/ForwardedByHeaderFilter.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.util.StringUtils;
+
+// ForwardedHeaderFilter does not parse the "by" part of the Forwarded header
+// This wraps the request so the local address/port are correct
+// https://github.com/spring-projects/spring-framework/issues/34654
+public class ForwardedByHeaderFilter implements Filter {
+
+  public static final Logger LOG = LoggerFactory.getLogger(ForwardedByHeaderFilter.class);
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+      throws IOException, ServletException {
+    if (request instanceof HttpServletRequest httpRequest) {
+      ServletRequest wrappedRequest = new ForwardedByHeaderRequest(httpRequest);
+      chain.doFilter(wrappedRequest, response);
+    } else {
+      chain.doFilter(request, response);
+    }
+  }
+
+  protected String formatRequest(HttpServletRequest request) {
+    return "HTTP " + request.getMethod() + " \"" + request.getRequestURI() + "\"";
+  }
+
+  // This is inspired by ForwardedHeaderExtractingRequest subclass of Spring Framework's
+  // ForwardedHeaderFilter
+  private static class ForwardedByHeaderRequest extends HttpServletRequestWrapper {
+
+    private static final String FORWARDED_VALUE = "\"?([^;,\"]+)\"?";
+    private static final Pattern FORWARDED_BY_PATTERN =
+        Pattern.compile("(?i:by)=" + FORWARDED_VALUE);
+    private final InetSocketAddress localAddress;
+
+    public ForwardedByHeaderRequest(HttpServletRequest servletRequest) {
+      super(servletRequest);
+      ServerHttpRequest request = new ServletServerHttpRequest(servletRequest);
+      HttpHeaders headers = request.getHeaders();
+      this.localAddress = parseForwardedBy(headers);
+    }
+
+    // This is adapted from Spring Framework's ForwardedHeaderUtils
+    public static InetSocketAddress parseForwardedBy(HttpHeaders headers) {
+      String forwardedHeader = headers.getFirst("Forwarded");
+      if (StringUtils.hasText(forwardedHeader)) {
+        String forwardedToUse = StringUtils.tokenizeToStringArray(forwardedHeader, ",")[0];
+        Matcher matcher = FORWARDED_BY_PATTERN.matcher(forwardedToUse);
+        if (matcher.find()) {
+          String value = matcher.group(1).trim();
+          int portSeparatorIdx = value.lastIndexOf(':');
+          int squareBracketIdx = value.lastIndexOf(']');
+          if (portSeparatorIdx > squareBracketIdx) {
+            if (squareBracketIdx == -1 && value.indexOf(':') != portSeparatorIdx) {
+              throw new IllegalArgumentException("Invalid IPv4 address: " + value);
+            }
+            String host = value.substring(0, portSeparatorIdx);
+            try {
+              int port = Integer.parseInt(value, portSeparatorIdx + 1, value.length(), 10);
+              return InetSocketAddress.createUnresolved(host, port);
+            } catch (NumberFormatException ex) {
+              throw new IllegalArgumentException(
```

Comment:
[non-blocking]

I'd be a bit nervous at throwing an unchecked exception.  Would this fail the whole request (perhaps with a 500 response) if the server fails to parse the `Forward` header value; and, if desired, wouldn't a 400 status code be better response?